### PR TITLE
Remove most of JPA related dependencies from auth service

### PR DESCRIPTION
**Pull request by petertrr**
* Remove `spring-data-configuration` from applied plugins it's not needed mostly; keep only depednecy needed to resolve one class
* Remove excludes on authoconfiguration classes that are no longer present

`auth-service` is a library and not a service so all Srping Boot starters are a bit redundant there
Related: #1408, #1425

### CODE REVIEW

File: authentication-service/build.gradle.kts [hunk 1]
```diff
@@ -25,6 +24,7 @@ dependencies {
     implementation(projects.saveCloudCommon)
     implementation(libs.spring.boot.starter.security)
     implementation(libs.spring.security.core)
+    implementation("org.springframework:spring-jdbc")
```
Comment (kgevorkyan) [inline authentication-service/build.gradle.kts:27]: Did you tried to start some of applications, that used `auth-service` as dependency?
I remember, that I tried to use such solution sometime ago, but got errors.
Does it all okay now?

Comment (kgevorkyan) [reply authentication-service/build.gradle.kts:27]: Build was successful in that case, but nor the `backend` not `sandbox` didn't started

Comment (petertrr) [reply authentication-service/build.gradle.kts:27]: Double checked, but I think for backend and sandbox it's safe - they have all spring-data related dependencies anyway. Also checked for gateway, which also has this dependency, and it's starting fine too.

**Comment by petertrr** [comment]
* Remove `spring-data-configuration` from applied plugins it's not needed mostly; keep only depednecy needed to resolve one class
* Remove excludes on authoconfiguration classes that are no longer present

**Pull request merged**